From d036186411768177d34c1cba39678eef98af8036 Mon Sep 17 00:00:00 2001
From: Hans Dedecker <dedeckeh@gmail.com>
Date: Sat, 18 Feb 2017 13:41:52 +0100
Subject: [PATCH 4/4] libubus: reset ctx->sock.error when doing ubus reconnect

When ubus connect fails ctx->sock.eof will be set but ctx->sock.error
can be set as well. Reset ctx->sock.error as well when doing ubus
reconnect

Signed-off-by: Hans Dedecker <dedeckeh@gmail.com>
(cherry picked from commit dfe338350e4dc6912a2552570fa5dd8d5bc84ff0)
---
 libubus-io.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libubus-io.c b/libubus-io.c
index 7e776cc..23782f9 100644
--- a/libubus-io.c
+++ b/libubus-io.c
@@ -369,6 +369,7 @@ int ubus_reconnect(struct ubus_context *ctx, const char *path)
 	}
 
 	ctx->sock.eof = false;
+	ctx->sock.error = false;
 	ctx->sock.fd = usock(USOCK_UNIX, path, NULL);
 	if (ctx->sock.fd < 0)
 		return UBUS_STATUS_CONNECTION_FAILED;
-- 
2.17.1

